plugins/newsql/newsql.c

```diff
@@ -2025,18 +2025,23 @@ void free_newsql_appdata(struct sqlclntstate *clnt)
     free(appdata->col_info.type);
 }
 
-extern void * (*externGetAuthData) (void*);
-extern void * (*externalMakeAuthData) (const char *, int, int, int, void *);
+extern void *(*externalMakeAuthData)(void *, const char *, int, int, int, void *);
 
-void * getAuthData(void * p_clnt) {
-    struct sqlclntstate *clnt = p_clnt;
+static void *newsql_get_authdata(struct sqlclntstate *clnt)
+{
     struct newsql_appdata *appdata = clnt->appdata;
     if (appdata) {
         CDB2SQLQUERY *sql_query = appdata->sqlquery;
-        if (sql_query && sql_query->identity) {
-            return externalMakeAuthData(sql_query->identity->principal, sql_query->identity->majorversion, sql_query->identity->minorversion, sql_query->identity->data.len, sql_query->identity->data.data);
+        if (sql_query && sql_query->identity && sql_query->identity->principal && sql_query->identity->data.data) {
```

Comment:
Perhaps this test - `&& sql_query->identity->principal && sql_query->identity->data.data)`

can also be moved into whatever implements `externalMakeAuthData`? Or am I missing something? When would identity be present but no principal/data?

Reply:
Its already check there, will remove the check from there.

Reply:
Removed it